packages/govern-server/src/core/data/index.ts

```diff
@@ -0,0 +1,127 @@
+  verbose?: boolean
+}
+
+function getSubgraphUrl(network: Network): string | null {
+  if (network.chainId === 1) {
+    return 'placeholder'
+  }
+  if (network.chainId === 4) {
+    return 'https://api.thegraph.com/subgraphs/name/0xgabi/aragon-govern-rinkeby-staging'
+  }
+  if (network.chainId === 100) {
+    return 'placeholder'
+  }
+  return null
+}
+
+class ConnectorTheGraph {
+  #gql: TheGraphWrapper
+  readonly config: ConnectorTheGraphConfig
+  readonly network: Network
+
+  constructor(config: ConnectorTheGraphConfig) {
+    this.config = config
+    this.network = toNetwork(config.network)
+
+    const orgSubgraphUrl = config.subgraphUrl || getSubgraphUrl(this.network)
+
+    if (!orgSubgraphUrl) {
+      throw new ErrorInvalidNetwork(
+        `The chainId ${this.network.chainId} is not supported by the TheGraph connector.`
+      )
+    }
+
+    this.#gql = new TheGraphWrapper(orgSubgraphUrl, {
+      verbose: config.verbose,
+    })
+  }
+
+  async dao(address: Address): Promise<QueryResult> {
+    try {
+      return this.#gql.performQuery(queries.DAO, {
+        dao: address.toLowerCase(),
+      })
+    } catch (err) {
+      throw new ErrorUnexpectedResult(
+        `Unexpected result when fetching the dao ${address}.`
+      )
+    }
+  }
+
+  async daos(): Promise<QueryResult> {
+    try {
+      return this.#gql.performQuery(queries.DAOS)
+    } catch (err) {
+      throw new ErrorUnexpectedResult(
+        `Unexpected result when fetching the daos.`
+      )
+    }
+  }
+
+  async queue(address: Address): Promise<QueryResult> {
+    try {
+      return this.#gql.performQuery(queries.QUEUE, {
+        queue: address.toLowerCase(),
+      })
+    } catch (err) {
+      throw new ErrorUnexpectedResult(
+        `Unexpected result when fetching the queue ${address}.`
+      )
+    }
+  }
+
+  async queues(): Promise<QueryResult> {
+    try {
+      return this.#gql.performQuery(queries.QUEUES)
+    } catch (err) {
+      throw new ErrorUnexpectedResult(
+        `Unexpected result when fetching the queue.`
+      )
+    }
+  }
+
+  async game(name: string): Promise<QueryResult> {
+    try {
+      return this.#gql.performQuery(queries.GAME, {
+        name,
+      })
+    } catch (err) {
+      throw new ErrorUnexpectedResult(
+        `Unexpected result when fetching the game ${name}.`
+      )
+    }
+  }
+
+  async games(): Promise<QueryResult> {
+    try {
+      return this.#gql.performQuery(queries.GAMES)
+    } catch (err) {
+      throw new ErrorUnexpectedResult(
+        `Unexpected result when fetching the games.`
+      )
+    }
+  }
+
+  async queuesForDao(dao: Address): Promise<QueryResult> {
+    try {
+      return this.#gql.performQuery(queries.QUEUES_BY_DAO, {
+        dao: dao.toString(),
+      })
+    } catch (err) {
+      throw new ErrorUnexpectedResult(
+        `Unexpected result when fetching the queues for dao ${dao}.`
+      )
+    }
+  }
+}
+
+export default ConnectorTheGraph
```

Comment:
Instead of having it at the end of the file could we also: ``export default class ConnectorTheGraph`` 

... not sure what the standard is :-)

Reply:
We usually export at the end of files, it makes it easy to spot and avoids splitting the class declaration over two lines :smile:

Reply:
This has to be a really really long class name lol.. But great to know! :)

... This means the export has to be changed for the other TS class. :-)

Reply:
> This has to be a really really long class name lol.. But great to know! :-)

haha yes, but I was more thinking of a full declaration with `implements` / `extends`, in which case it can exceed 80 chars.